### ISSUE TITLE
SARIF: Please always include `results: []` property - required value by GitHub

### ISSUE THREAD

**Issue by jamacku** [csutils/csdiff#102]
When no defects are supplied to csgrep, the produced SARIF file is missing the `results` property. But this property is required by GitHub, therefore when trying to upload such a SARIF, it results in an error.

[GitHub Documentation](https://docs.github.com/en/code-security/code-scanning/integrating-with-code-scanning/sarif-support-for-code-scanning#run-object):

_`results[]` - **Required**. The results of the analysis tool. Code scanning displays the results on GitHub. For more information, see the [result object](https://docs.github.com/en/code-security/code-scanning/integrating-with-code-scanning/sarif-support-for-code-scanning#result-object)._

Encountered in:
- https://github.com/fedora-sysv/initscripts/pull/449
- workflow run - https://github.com/fedora-sysv/initscripts/actions/runs/4043571776

[SARIF file](https://github.com/csutils/csdiff/files/10535733/output.sarif.log) produced by `csgrep`:

```json
{
    "$schema": "https://json.schemastore.org/sarif-2.1.0.json",
    "version": "2.1.0",
    "inlineExternalProperties": [
        {
            "externalizedProperties": {
                "tool": "vcs-diff-lint",
                "tool-url": "https://github.com/fedora-copr/vcs-diff-lint#readme"
            }
        }
    ],
    "runs": [
        {
            "tool": {
                "driver": {
                    "name": "vcs-diff-lint",
                    "informationUri": "https://github.com/fedora-copr/vcs-diff-lint#readme"
                }
            }
        }
    ]
}
```

**Comment by kdudka** [comment]
Thank you for reporting it!  This should be a one-liner...

**Comment by kdudka** [comment]
Actually, the one-liner fixed it only partially.  The underlying JSON encoder implemented in Boost encodes it as `"results" : ""` rather than `"results" : []`: https://github.com/csutils/csdiff/pull/104/files#diff-4c1b6ef70f98eb9c11098a4dbfba18bb98b441b20ee57d0ab83e017bf3df0e1bR20

Is such a change sufficient for GitHub to accept the data?

**Comment by lzaoral** [comment]
Note, that Boost-JSON does not have this issue apart from being way too new for EPEL.

**Comment by jamacku** [comment]
Unfortunately, `github/codeql-action/upload-sarif` explicitly checks for an array. Please see the following [snippet](https://github.com/github/codeql-action/blob/4827181940cfa69fb1a00e1d7a2de521f866f3f6/src/upload-lib.ts#L216-L223):

```ts
for (const run of parsedSarif.runs) {
  if (!Array.isArray(run.results)) {
    throw new Error("Invalid SARIF. Missing 'results' array in run.");
  }
  numResults += run.results.length;
}
```

**Comment by kdudka** [comment]
@jamacku Thanks for the quick reply on this!  I will prepare a better pull request with a quick and dirty solution after #103 is merged.

> Note, that Boost-JSON does not have this issue apart from being way too new for EPEL.

@lzaoral Good point.  That would be the ideal long term solution to problems like this.

**Comment by lzaoral** [comment]
This issue will be completely resolved by #105 which replaces Kamil's hacky workaround with a proper JSON serializer.
